crates/rooch-framework/sources/brc20.move

```diff
@@ -0,0 +1,384 @@
+// Copyright (c) RoochNetwork
+// SPDX-License-Identifier: Apache-2.0
+
+module rooch_framework::brc20 {
+    //use std::vector;
+    use std::option::{Self, Option};
+    use std::string::{Self, String};
+    use moveos_std::json;
+    use moveos_std::context::{Self, Context};
+    use moveos_std::object::{Self, Object};
+    use moveos_std::table::{Self, Table};
+    use moveos_std::table_vec;
+    use moveos_std::simple_map::{Self, SimpleMap};
+    use moveos_std::string_utils;
+    use rooch_framework::bitcoin_address::{BTCAddress};
+    use rooch_framework::ord::{Self, Inscription, InscriptionStore};
+    // use rooch_framework::bitcoin_types::{Self, Witness, Transaction};
+    // use rooch_framework::bitcoin_light_client::{Self, BitcoinBlockStore};
+
+    friend rooch_framework::genesis;
+
+    //TODO should we register the BRC20 as a CoinInfo?
```

Comment:
In the Indexer extension, specific types of Objects can be parsed, including `CoinInfo` and `CoinStore`. At the same time, by linking the Account, we can get the Coin-dimensional Indexer.

If Bitcoin and Ethereum can reuse `CoinInfo`, a full-chain Coin Indexer can be generated.

The generation of NFT Indexers may be more complicated.